internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/BlobP2PTransferAmongServersTest.java

```diff
@@ -0,0 +1,163 @@
+import com.linkedin.venice.controllerapi.UpdateStoreQueryParams;
+import com.linkedin.venice.integration.utils.ServiceFactory;
+import com.linkedin.venice.integration.utils.VeniceClusterWrapper;
+import com.linkedin.venice.integration.utils.VeniceServerWrapper;
+import com.linkedin.venice.meta.PersistenceType;
+import com.linkedin.venice.offsets.OffsetRecord;
+import com.linkedin.venice.store.rocksdb.RocksDBUtils;
+import com.linkedin.venice.utils.IntegrationTestPushUtils;
+import com.linkedin.venice.utils.TestUtils;
+import com.linkedin.venice.utils.TestWriteUtils;
+import com.linkedin.venice.utils.Utils;
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Properties;
+import java.util.function.Consumer;
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class BlobP2PTransferAmongServersTest {
+  private static final Logger LOGGER = LogManager.getLogger(BlobP2PTransferAmongServersTest.class);
+
+  private String path1;
+  private String path2;
+
+  @Test(singleThreaded = true)
+  public void testBlobP2PTransferAmongServers() throws Exception {
+    // set up store with 2 servers and trigger the job
+    VeniceClusterWrapper cluster = initializeVeniceCluster();
+    String storeName = "test-store";
+    Consumer<UpdateStoreQueryParams> paramsConsumer = params -> params.setBlobTransferEnabled(true);
+    setUpStore(cluster, storeName, paramsConsumer, properties -> {}, true);
+
+    VeniceServerWrapper server1 = cluster.getVeniceServers().get(0);
+    VeniceServerWrapper server2 = cluster.getVeniceServers().get(1);
+
+    // very the snapshot is generated for both servers after the job is done
+    for (int i = 0; i < 3; i++) {
+      String snapshotPath1 = RocksDBUtils.composeSnapshotDir(path1 + "/rocksdb", storeName + "_v1", i);
+      Assert.assertTrue(Files.exists(Paths.get(snapshotPath1)));
+      String snapshotPath2 = RocksDBUtils.composeSnapshotDir(path2 + "/rocksdb", storeName + "_v1", i);
+      Assert.assertTrue(Files.exists(Paths.get(snapshotPath2)));
+    }
+
+    // cleanup and restart server 1
+    FileUtils.deleteDirectory(
+        new File(RocksDBUtils.composePartitionDbDir(path1 + "/rocksdb", storeName + "_v1", METADATA_PARTITION_ID)));
+    FileUtils.deleteDirectory(new File(RocksDBUtils.composePartitionDbDir(path1 + "/rocksdb", storeName + "_v1", 0)));
+    FileUtils.deleteDirectory(new File(RocksDBUtils.composePartitionDbDir(path1 + "/rocksdb", storeName + "_v1", 1)));
+    FileUtils.deleteDirectory(new File(RocksDBUtils.composePartitionDbDir(path1 + "/rocksdb", storeName + "_v1", 2)));
+    cluster.stopAndRestartVeniceServer(server1.getPort());
+
+    // wait 2 min
+    Thread.sleep(120000);
+
+    // the partition files should be transferred to server 1 and offset should be the same
```

Comment:
The concern is we don't know whether the server ingests the data via blob transfer or the kafka.  Yes, the logging can help us here but it's not a programmatic way and the chance is someone breaks the feature in the future and we don't know.

We discussed this in [another thread](https://github.com/linkedin/venice/pull/1118#discussion_r1719109319) a while ago. Do you think we can leverage that to validate?

Reply:
discussed offline. The IDENTITY file changes so it cannot be used to verify whether partitions really happened or not. There's no reason to block this PR at least and we will come back to the topic later on.